shared/route-tree/index.js

```diff
@@ -151,11 +151,11 @@ export function routeSetProps (routeDef: RouteDefNode, routeState: ?RouteStateNo
   const parentPathSeq = I.Seq(parentPath || []).map(item => {
     return {type: 'traverse', next: item}
   })
-  return _routeSet(routeDef, routeState, parentPathSeq.concat(pathSeq))
+  return _routeSet(routeDef, routeState, parentPathSeq.concat(pathSeq), persistState)
 }
 
-export function routeNavigate (routeDef: RouteDefNode, routeState: ?RouteStateNode, pathProps: PathParam<*>, parentPath: ?Path): RouteStateNode {
-  return routeSetProps(routeDef, routeState, I.List(pathProps).push({selected: null, props: {}}), parentPath)
+export function routeNavigate (routeDef: RouteDefNode, routeState: ?RouteStateNode, pathProps: PathParam<*>, parentPath: ?Path, persistState: ?boolean = false): RouteStateNode {
+  return routeSetProps(routeDef, routeState, I.List(pathProps).push({selected: null, props: {}}), parentPath, !!persistState)
```

Comment:
:sheep: Is the `!!` necessary? The argument type is boolean, shouldn't that suffice?